src/cli/onefuzz/templates/regression.py

```diff
@@ -0,0 +1,250 @@
+
+        if not ((crashes and input_reports) or inputs):
+            self.logger.error(
+                "please specify either the 'crash' and 'input_reports' parameters or the inputs parameter"
+            )
+
+        if dryrun:
+            return None
+
+        self.logger.info("creating regression task from template")
+
+        helper = JobHelper(
+            self.onefuzz,
+            self.logger,
+            project,
+            name,
+            build,
+            duration,
+            pool_name=pool_name,
+            target_exe=target_exe,
+        )
+
+        helper.add_tags(tags)
+        if crashes:
+            if self.onefuzz.containers.get(crashes):
+                helper.define_containers(ContainerType.unique_inputs)
+            else:
+                self.logger.error(f"invalid crash container {crashes}")
+
+        if input_reports:
+            if self.onefuzz.containers.get(input_reports):
+                helper.define_containers(ContainerType.input_reports)
+            else:
+                self.logger.error(f"invalid crash container {input_reports}")
+
+        if inputs:
+            helper.define_containers(ContainerType.readonly_inputs)
+
+        helper.define_containers(
+            ContainerType.setup,
+            ContainerType.reports,
+            ContainerType.no_repro,
+        )
+
+        helper.create_containers()
+        helper.setup_notifications(notification_config)
+        if inputs:
+            helper.upload_inputs(inputs, read_only=True)
+
+        containers = [
+            (ContainerType.setup, helper.containers[ContainerType.setup]),
+            (ContainerType.reports, helper.containers[ContainerType.reports]),
+            (ContainerType.no_repro, helper.containers[ContainerType.no_repro]),
+        ]
+
+        if crashes:
+            containers.append((ContainerType.crashes, crashes))
+        if input_reports:
+            containers.append((ContainerType.input_reports, input_reports))
+
+        helper.upload_setup(setup_dir, target_exe)
+        target_exe_blob_name = helper.target_exe_blob_name(target_exe, setup_dir)
+
+        self.logger.info("creating regression task")
+        regression_task = self.onefuzz.tasks.create(
+            helper.job.job_id,
+            task_type,
+            target_exe_blob_name,
+            containers,
+            pool_name=pool_name,
+            duration=duration,
+            vm_count=1,
+            reboot_after_setup=reboot_after_setup,
+            target_options=target_options,
+            target_env=target_env,
+            tags=tags,
+            target_timeout=crash_report_timeout,
+            check_retry_count=check_retry_count,
+            debug=debug,
+            check_fuzzer_help=check_fuzzer_help,
+            report_list=report_list,
+        )
+        helper.wait_for_stopping = fail_on_repro
+
+        self.logger.info("done creating tasks")
+        helper.wait()
+        if fail_on_repro:
```

Comment:
what does this do in the case that the task fails?

It would be useful to differentiate task failure from report files not being created.

Reply:
added a case for when we detect a job error